test/ui/test_reduce_and_export_data.py

```diff
@@ -0,0 +1,147 @@
+from RefRed.main import MainGui
+from qtpy import QtCore, QtWidgets
+import os
+from unittest import mock
+import numpy as np
+
+wait = 200
+
+
+@mock.patch("qtpy.QtWidgets.QFileDialog")
+def test_reduce_and_export_data(QFileDialog_mock, qtbot, tmp_path, data_server):
+    # set mock return values for QFileDialog
+    QFileDialog_mock().exec_.return_value = True
```

Comment:
Can you replace lines 13 through 39 using fixtures `data_server` and  `main_gui`
```python
window = main_gui(configuration=data_server.path_to("REF_L_188298_auto_template.xml"),
                  show=True)
```

Reply:
Using this fixture doesn't work if I mock QFileDialog.getSaveFileName and QFileDialog.getExistingDirectory which means I am going to rewrite how I am handling them. I don't have time this week for such a large refactor so this will have to wait until next week.

Reply:
I haven't noticed that you had mocked method `getSaveFileName()` for later use when saving the ascii file. I'll approve the PR as it is, then